docs/changelog.rst

```diff
@@ -21,11 +21,47 @@ Changes:
   ``dec_ref()`` are now enabled by default again.
   `#4246 <https://github.com/pybind/pybind11/pull/4246>`_
 
+* ``py::initialize_interpreter()`` using ``PyConfig_InitPythonConfig()``
+  instead of ``PyConfig_InitIsolatedConfig()``, to obtain complete
+  ``sys.path``.
+  `#4473 <https://github.com/pybind/pybind11/pull/4473>`_
+
+* Cast errors now always include Python type information, even if
+  ``PYBIND11_DETAILED_ERROR_MESSAGES`` is not defined. This increases binary
+  sizes slightly (~1.5%) but the error messages are much more informative.
+  `#4463 <https://github.com/pybind/pybind11/pull/4463>`_
+
+
 Build system improvements:
 
 * Update clang-tidy to 15 in CI.
   `#4387 <https://github.com/pybind/pybind11/pull/4387>`_
 
+* Moved the linting framework over to Ruff.
+  `#4483 <https://github.com/pybind/pybind11/pull/4483>`_
+
+Version 2.10.4 (Mar 16, 2023)
+----------------------------
+
+Changes:
+
+* ``python3 -m pybind11`` gained a ``--version`` option (prints the version and
+  exits).
+  `#4526 <https://github.com/pybind/pybind11/pull/4526>`_
+
+Bug Fixes:
+
+* Fix a warning when pydebug is enabled on Python 3.11.
```

Comment:
I think the changelog message here is slightly misleading, since this issue could cause undefined behavior with GC-tracked objects besides just issuing a warning.

Reply:
How would you word this?

Also, do you think #4461 was fine? I've been a bit uneasy about the change because we don't even have a unit test for it. See long discussion under #4461.